### PR TITLE
feat(system): Accept-header JSON content negotiation (#2663)

### PR DESCRIPTION
**Pull request by natechadwick-intsof**
## Summary

Implements **Accept-header JSON content negotiation** for classic XML Application responses (parent residual of #2662 / baseline #2660).

When the request page has **no extension**, honor `Accept: application/json` (and structured `application/*+json`) so the response uses `PAGE_TYPE_JSON` / `application/json`. Precedence:

1. **Known extension** always wins (`.xml` / `.html` / `.htm` / `.txt` / `.json`)
2. **No extension** + Accept prefers JSON → JSON
3. Otherwise → product default **XML**

Does **not** change behavior when Accept is missing, prefers XML/HTML, or has equal quality for JSON and XML. Unknown extensions stay `PAGE_TYPE_UNKNOWN` (not negotiated).

### Changes
- `system/.../PSRequest.java` — Accept negotiation in `setRequestFileURL`; safe wrapper cast; bind servlet request before URL parse so headers are available
- `PSRequestTest` — negotiation order / extension-wins / quality-factor cases
- `docs/developer-module/classic-xml-app-json-io.md` — document precedence

## Parent tracker
Part of #2662 (Classic XML Application JSON I/O residuals).

## Test plan
- [x] `cd system && ../mvnw.cmd clean install` — **BUILD SUCCESS**
- [x] `PSRequestTest` — Tests run: 13, Failures: 0
- [x] Full system surefire — Tests run: 1447, Failures: 0, Errors: 0, Skipped: 237
- [x] No new warnings attributable to this change (baseline javadoc/dependency-analyze noise only)

## Gates evidence
- Pre-PR Maven: standalone `cd system; ..\mvnw.cmd clean install` → BUILD SUCCESS
- Erlang-style self-review: no bugs; portable (no path I/O); companions = unit tests + feature doc; extension precedence preserved
- Operator: Grok: night-issue-prs (model grok-4.5)

Fixes #2663
Refs #2662

> Co-Authored by Grok Build using grok-4.5 with agent main.

### CODE REVIEW

File: system/src/main/java/com/percussion/server/PSRequest.java [hunk 1]
```diff
@@ -548,7 +566,111 @@ public void setRequestFileURL(String reqFileURL) {
...
+      int semi = range.indexOf(';');
+      if (semi >= 0) {
+        mediaType = range.substring(0, semi).trim();
+        String params = range.substring(semi + 1);
+        StringTokenizer paramTok = new StringTokenizer(params, ";");
+        while (paramTok.hasMoreTokens()) {
+          String param = paramTok.nextToken().trim();
+          if (param.regionMatches(true, 0, "q=", 0, 2)) {
+            try {
+              q = Double.parseDouble(param.substring(2).trim());
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply system/src/main/java/com/percussion/server/PSRequest.java]: **Mitigation (commit `28aaa50cd5`):** Clamped Accept q-values to `[0, 1]` after parse (`Math.min(1.0, Math.max(0.0, …))`) per RFC 7231 so `application/json;q=1.5` cannot outrank equal-quality XML/HTML types. Covered by `testAcceptJsonQAboveOneClampedDoesNotBeatDefaultXml`. `PSRequestTest`: 17 tests, 0 failures.

> Co-Authored by Grok CLI using grok-4.20 with agent overnight-pr-followup.

File: system/src/test/java/com/percussion/server/PSRequestTest.java [hunk 2]
```diff
@@ -109,6 +109,155 @@ public void testJsonPageTypeFromExtension() {
...
+    assertFalse(PSRequest.isJsonAcceptMediaType("application/xml"));
+    assertFalse(PSRequest.isJsonAcceptMediaType("*/*"));
+
+    assertTrue(PSRequest.isXmlOrHtmlAcceptMediaType("application/xml"));
+    assertTrue(PSRequest.isXmlOrHtmlAcceptMediaType("text/xml"));
+    assertTrue(PSRequest.isXmlOrHtmlAcceptMediaType("text/html"));
+    assertTrue(PSRequest.isXmlOrHtmlAcceptMediaType("application/xhtml+xml"));
+    assertTrue(PSRequest.isXmlOrHtmlAcceptMediaType("application/atom+xml"));
+    assertFalse(PSRequest.isXmlOrHtmlAcceptMediaType("application/json"));
+  }
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply system/src/test/java/com/percussion/server/PSRequestTest.java:320]: **Mitigation (commit `28aaa50cd5`):** Added Accept edge-case tests: `testAcceptStarStarAloneDefaultsToXml` (`*/*` → XML), `testAcceptJsonQZeroDoesNotSelectJson`, `testAcceptXmlQZeroJsonSelected`, and `testAcceptJsonQAboveOneClampedDoesNotBeatDefaultXml` (q clamp + equal-quality XML). `PSRequestTest`: 17 tests, 0 failures.

> Co-Authored by Grok CLI using grok-4.20 with agent overnight-pr-followup.

**Review by natechadwick — APPROVED** [review]
